missioncontrol/home/migrations/0021_s3files_20190227_0220.py

```diff
@@ -0,0 +1,48 @@
+# Generated by Django 2.1.4 on 2019-02-27 03:37
+
+from django.db import migrations, models
+import django.db.models.deletion
+import django.utils.timezone
+import home.models
+import uuid
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('home', '0020_merge_20190124_0028'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='S3File',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('cid', models.CharField(max_length=128, unique=True)),
+                ('bucket', models.CharField(max_length=64)),
+                ('what', models.TextField(default='default')),
+                ('created', models.DateTimeField(default=django.utils.timezone.now)),
+                ('where', models.TextField(default='nowhere')),
+            ],
+            bases=(models.Model, home.models.Serializable),
+        ),
+        migrations.CreateModel(
+            name='TaskRun',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('uuid', models.UUIDField(default=uuid.uuid4, unique=True)),
+                ('task', models.TextField()),
+                ('start_time', models.DateTimeField()),
+                ('end_time', models.DateTimeField()),
+                ('exit_code', models.IntegerField()),
+                ('task_pass', models.ForeignKey(db_column='pass', on_delete=django.db.models.deletion.PROTECT, to='home.Pass', to_field='uuid')),
```

Comment:
what's the use case for this field?

Reply:
A task run will always be associated with a pass (currently at least)
So say you want to find all tasks run during a specific pass, you query them (in django at least) with `TaskRun.objects.filter(task_pass=<uuid>)` or from the api `/passes/{pass_uuid}/task-runs/` 

You actually made me notice a bug that was returning all task runs instead of just a passes ones, that's lucky :tada:

Reply:
Oh man, I totally spaced on this - sorry! I was thinking that you'd look it up by TaskStack, but those are immutable and can spawn many TaskRuns, so this totally makes sense. :+1: